src/eos-updater-avahi.c

```diff
@@ -180,6 +184,31 @@ queue_error_callback (EosAvahiDiscoverer *discoverer,
 }
 
 static void
+maybe_queue_error_callback (EosAvahiDiscoverer *discoverer,
+                            const gchar *format,
+                            ...)
+{
+  va_list args;
+
+  if (discoverer->state != EOS_AVAHI_RESOLVING_ONLY)
+    return;
+
+  if (discoverer->callback_id > 0)
+    return;
+
+  if (g_hash_table_size (discoverer->discovered_services) > 0)
+    return;
```

Comment:
These three conditions already exist in `maybe_queue_success_callback()`. If you refactor them in e.g. a `should_queue_callback()` function, this one could be just implemented as the composition of that and `queue_error_callback()`.